src/www/views/servers_view/server_list_item/server_card/index.ts

```diff
@@ -134,7 +134,7 @@ const getSharedComponents = (element: ServerListItemElement & LitElement) => {
   const hasErrorMessage = Boolean(server.errorMessageId);
 
   const messages = {
-    serverName: server.name ?? localize(server.isOutlineServer ? 'server-default-name-outline' : 'server-default-name'),
+    serverName: server.name || localize(server.isOutlineServer ? 'server-default-name-outline' : 'server-default-name'),
```

Comment:
this shows the default server name if the string is empty as well as undefined.

Reply:
That's fine. But why would the name be empty?

Reply:
Sorry, I'm not sure what you're asking - I believe the server name comes from the ss:// fragment? So it very easily could be omitted. If you mean why it would be an empty _string_ -  it seems that's what the `SHADOWSOCKS_URI.parse` returns https://github.com/Jigsaw-Code/outline-client/blob/04290eb01a730433c97a4835a583469ece1318cb/src/www/app/outline_server.ts#L321-L322

Reply:
<img width="926" alt="Screen Shot 2022-06-27 at 17 31 32" src="https://user-images.githubusercontent.com/3759828/176047049-62ece2a7-9aa5-4840-ba70-b8944394e367.png">

Reply:
Shall we also treat pure space (`.trim()`) strings as empty strings as well? Maybe not a priority for now.

Reply:
I can add a .trim(), that's easy

Reply:
`server.name.trim()` will trigger NullException when `server.name` is `null`. Need to be something like `server.name?.trim() || ...`

Reply:
Fixed!